src/ActionServer_FJT.c

```diff
@@ -494,26 +630,79 @@ void Ros_ActionServer_FJT_Goal_Complete(GOAL_END_TYPE goal_end_type)
+            &feedback_FollowJointTrajectory.feedback.joint_names,
+            posTolerance, numAxesToCheck);
+
+        BOOL bToleranceParseOk = TRUE;
+        if (statusParseGoalTolerance != OK)
+        {
+            Ros_Debug_BroadcastMsg("%s: parsing 'goal_tolerance' field failed: %d", __func__, statusParseGoalTolerance);
+            //TODO(gavanderhoorn): implement tolerance parsing in Ros_ActionServer_FJT_Goal_Received(..) instead
+            bToleranceParseOk = FALSE;
+            Ros_Debug_BroadcastMsg("%s: skipping goal tolerance check", __func__);
+            goto goal_complete_skip_tolerance_comparison;
+        }
+
+        //retrieve the last traj pt from the goal traj and re-order position values such
+        //that they correspond to the internal MotoROS2 ordering (as used in
+        //feedback_FollowJointTrajectory.feedback)
+        double lastTrajPtPositions[MR2_JTA_MAX_NUM_AXES];
+        bzero(lastTrajPtPositions, sizeof(lastTrajPtPositions));
+        size_t finalTrajPtIdx = g_actionServer_FJT_SendGoal_Request.goal.trajectory.points.size - 1;
+
+        STATUS statusGoalToleranceReorder = Ros_ActionServer_FJT_Reorder_TrajPt_To_Internal_Order(
+            &g_actionServer_FJT_SendGoal_Request.goal.trajectory.points.data[finalTrajPtIdx],
+            &g_actionServer_FJT_SendGoal_Request.goal.trajectory.joint_names,
+            &feedback_FollowJointTrajectory.feedback.joint_names,
+            lastTrajPtPositions,
+            numAxesToCheck);
+
+        if (statusGoalToleranceReorder != OK)
         {
-            diff = fabs(feedback_FollowJointTrajectory.feedback.desired.positions.data[axis] - feedback_FollowJointTrajectory.feedback.actual.positions.data[axis]);
+            Ros_Debug_BroadcastMsg("%s: reordering final traj pt failed: %d", __func__, statusGoalToleranceReorder);
+            //TODO(gavanderhoorn): check re-ordering in Ros_ActionServer_FJT_Goal_Received(..) instead
+            bToleranceParseOk = FALSE;
+            Ros_Debug_BroadcastMsg("%s: skipping goal tolerance check", __func__);
+            goto goal_complete_skip_tolerance_comparison;
+        }
 
-            double posTolerance = g_actionServer_FJT_SendGoal_Request.goal.goal_tolerance.data[axis].position; //user-provided a goal_tolerance
-            if (posTolerance == 0.0) //user did NOT provide a tolerance
-                posTolerance = DEFAULT_FJT_GOAL_POSITION_TOLERANCE;
 
-            if (diff > posTolerance)
+        //compare current state with target positions from last traj point
+        for (int axis = 0; axis < numAxesToCheck; axis += 1)
+        {
+            double current_jstate = feedback_FollowJointTrajectory.feedback.actual.positions.data[axis];
+            diff = fabs(lastTrajPtPositions[axis] - current_jstate);
+            Ros_Debug_BroadcastMsg("desired - actual: %12.8f - %12.8f: %12.8f",
+                lastTrajPtPositions[axis], current_jstate, diff);
```

Comment:
```suggestion
            Ros_Debug_BroadcastMsg("target pos (%12.8f) - actual pos (%12.8f) = (%12.8f)",
                lastTrajPtPositions[axis], current_jstate, diff);
```

Reply:
Also, I'm getting zeroes again in the output (just for `lastTrajPtPositions[axis]`. I have no idea why

Reply:
Actually, as I look closer, none of the values make sense.

```
[1715980487.96850944] [192.168.1.31:50399]: 2024-05-17 21:14:47.967896 desired - actual:   0.00000000 -   0.00046206:   0.00046206
[1715980487.96860313] [192.168.1.31:50399]: 2024-05-17 21:14:47.967896 desired - actual:   0.00000000 -  -0.00001342:   0.00001342
[1715980487.96866250] [192.168.1.31:50399]: 2024-05-17 21:14:47.967896 desired - actual:   0.00000000 -   0.00001227:   0.00001227
[1715980487.96870780] [192.168.1.31:50399]: 2024-05-17 21:14:47.968096 desired - actual:   0.00000000 -   0.00000000:   0.00000000
[1715980487.96891475] [192.168.1.31:50399]: 2024-05-17 21:14:47.968096 desired - actual:   0.00000000 -   0.00040952:   0.00040952
[1715980487.96918893] [192.168.1.31:50399]: 2024-05-17 21:14:47.968496 desired - actual:   0.00000000 -   0.00000000:   0.00000000

```

Reply:
> Also, I'm getting zeroes again in the output (just for `lastTrajPtPositions[axis]`. I have no idea why

me neither.

I was under the impression I'd 'fixed' that when referring to joint names, but apparently you still get them.

PS: you really need to update your `debug_listener.py` checkout.